src/library/dlgtagfetcher.cpp

```diff
@@ -147,18 +192,47 @@ void DlgTagFetcher::loadTrack(const QModelIndex& index) {
 
 void DlgTagFetcher::slotTrackChanged(TrackId trackId) {
     if (m_track && m_track->getId() == trackId) {
-        updateStack();
+        // Track has just updated with the suggested tag.
+        // Original tag needs to be updated as well.
+        tags->clear();
+        addDivider(tr("Original tags"), tags);
+        addTrack(trackColumnValues(*m_track), kOriginalTrackIndex, tags);
+
+        addDivider(tr("Suggested tags"), tags);
```

Comment:
I tried that before a week ago or something, that didn't worked at that time. Because tags were displayed and editing their data wasn't changing their appearance on the QTreeItem.

Now I tried it again, and changing their text actually worked. No need to update the whole tags again IMHO 👍